autoextract_poet/page_inputs.py

```diff
@@ -0,0 +1,59 @@
+from typing import ClassVar, List, Optional, Type
+
+import attr
+
+from autoextract_poet.items import (
+    Article,
+    Item,
+    Product,
+)
+
+
+@attr.s(auto_attribs=True)
+class ResponseData:
+    """Container for AutoExtract response data.
+
+    Should not be used directly by providers.
+    Use derived classes like ArticleResponseData and similar.
+
+    API responses are wrapped in a JSON array
+    (this is to facilitate query batching).
+
+    https://doc.scrapinghub.com/autoextract.html#responses
+    """
+
+    item_class: ClassVar[Type[Item]]
+    item_key: ClassVar[str]
+
+    data: dict
+
+    def to_items(self) -> Optional[List[Item]]:
```

Comment:
It seems `from_list` always return a `dict`, so it shouldn't be optional.
Or is there some other use case?

Reply:
Also, in the other PR we had a conversation on the return type.
So, I guess it follows here, right?
Since this is the base class, it should've no return type as you can't infer from the base classes.
Even though, a product is an item, in the other it was opted out for the many complications it could cause.

Reply:
> Even though, a product is an item, in the other it was opted out for the many complications it could cause.

This. I don't think we'll need to override this method.